js/src/path.js

```diff
@@ -40,13 +40,16 @@ export interface Part {
 export default class Path {
   _parts: Array<Part>;
 
-  static parse(str: string): Path {
+  /**
+   * Returns `str` parsed as Path if successful, or a null path with an error message if not.
+   */
+  static parse(str: string): [Path | null, string /* error */] {
```

Comment:
I decided to change this to not throw errors... it's a bit weird for a JS idiom, but it's either that or adding a separate way to validate paths (and likewise absolute paths). Validation via try to parse --> catch exception isn't good.